### ISSUE TITLE
[0.15.2 /  0.4.0Beta] NPE caused by muffling mod

### ISSUE THREAD

**Issue by AlienXAXS** [korti11/bettermuffling#10]
Hi,
My game crashed with the following stack trace:

```
java.lang.NullPointerException: Unexpected error
	at io.korti.bettermuffling.client.sound.SoundHandler.lambda$onSoundPlaying$0(SoundHandler.java:29) ~[?:0.4.0-Beta] {re:classloading}
	at io.korti.bettermuffling.client.sound.SoundHandler$$Lambda$18864/413025477.accept(Unknown Source) ~[?:?] {}
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:1.8.0_51] {}
	at io.korti.bettermuffling.client.sound.SoundHandler.onSoundPlaying(SoundHandler.java:25) ~[?:0.4.0-Beta] {re:classloading}
	at net.minecraftforge.eventbus.ASMEventHandler_1189_SoundHandler_onSoundPlaying_PlaySoundEvent.invoke(.dynamic) ~[?:?] {}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80) ~[eventbus-2.1.0-service.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-2.1.0-service.jar:?] {}
	at net.minecraftforge.client.ForgeHooksClient.playSound(ForgeHooksClient.java:353) ~[?:?] {re:classloading}
	at mekanism.client.sound.SoundHandler$TileTickableSound.func_73660_a(SoundHandler.java:228) ~[?:?] {re:classloading}
	at net.minecraft.client.audio.SoundEngine.func_217927_h(SoundEngine.java:176) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.audio.SoundEngine.func_217921_a(SoundEngine.java:164) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.audio.SoundHandler.func_215290_a(SoundHandler.java:182) ~[?:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1408) ~[?:?] {re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:904) [?:?] {re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:553) [?:?] {re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(SourceFile:204) [?:?] {re:classloading}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:56) [forge-1.15.2-31.1.63.jar:31.1] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$470/1934424193.call(Unknown Source) [forge-1.15.2-31.1.63.jar:31.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-5.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-5.1.0.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-5.1.0.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81) [modlauncher-5.1.0.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65) [modlauncher-5.1.0.jar:?] {re:classloading}
```

**What happened before the crash**
I was very far from home, over 10,000 blocks away while being attacked by Vex mobs in a Mansion.
I teleported back home using /home, and my game instantly crashed with the above stack - we have a muffler block at home.

I wonder if the muffler block attempted to iterate over (as per the forEach loop before it) a mob that was disposed of during the teleport, or a mob that my client had not loaded in correctly yet due to chunks loading in still.

Thought you should know, perhaps add some null checks in this area of the code to prevent crashing in the future.

Cheers :)

**Comment by AlienXAXS** [comment]
Just had exactly the same crash when Teleporting home from the end after killing the ender dragon (not that i would expect the action of killing him actually made it crash!)

**Comment by korti11** [comment]
Should be fixed in the new version 0.5.0-Beta and is ready to download on the curse page.
